engine/lambda-functions/notify-run-result/handler_test.go

```diff
@@ -163,15 +155,36 @@ func TestNotifyRunResultHandler_Provisioning_Success(t *testing.T) {
 		StatusTimestamps:     nil,
 	})
 
-	testStateVersionOutput := tfe.StateVersionOutput{
+	testStateVersionOutput := &tfe.StateVersionOutput{
 		Name:      "super_valuable_information_about_your_infra",
 		Sensitive: true,
 		Type:      "string",
 		Value:     "supervaluableinfo",
 	}
 
-	tfcServer.AddStateVersion(testWorkspace.ID, &tfe.StateVersion{
-		Outputs: []*tfe.StateVersionOutput{&testStateVersionOutput},
+	// Add a state version that contains no outputs
+	tfcServer.AddStateVersion(testApply.ID, &tfe.StateVersion{
+		Outputs:   []*tfe.StateVersionOutput{{}},
+		CreatedAt: time.Now().Add(-time.Minute * 2),
+	})
+
+	// Add a more recent state version that contains the output
+	tfcServer.AddStateVersion(testApply.ID, &tfe.StateVersion{
+		Outputs:   []*tfe.StateVersionOutput{testStateVersionOutput},
+		CreatedAt: time.Now().Add(-time.Minute * 1),
+	})
+
+	// Add an even older state version (this will help ensure that we are testing that state versions are sorted by
+	// CreatedAt in order to find the most recent state version
```

Comment:
**Non-blocking suggestion:** We need a closing parenthesis in this comment or we should remove the opening one:
```suggestion
	// CreatedAt in order to find the most recent state version)
```